AmazonChimeSDK/AmazonChimeSDKTests/audiovideo/video/backgroundfilter/BackgroundFilterTest.swift

```diff
@@ -120,8 +171,37 @@ class BackgroundFilterTests: XCTestCase {
             }
             processor.onVideoFrameReceived(frame: frame)
 
-            // Verify checksum is as expected.
-            XCTAssert(expectedReplacementHashes[index].contains(hash), "Failed with \(hash) received hash.")
+            // Wait for the image to generate before proceeding to avoid non-determinism.
+            waitForExpectations(timeout: 1) { error in
+                if let error = error {
+                    XCTFail("waitForExpectationsWithTimeout errored for index \(index): \(error)")
+                }
+            }
+
+            guard let gotCgImage = self.resize(image: processedImage!, to: downscaledSize).cgImage,
+                let gotCgImageData = gotCgImage.dataProvider?.data,
+                let gotCgImageBytes = CFDataGetBytePtr(gotCgImageData) else {
+                XCTFail("Couldn't access CGImage data")
+                return
+            }
+            guard let expectedCgImage = self.resize(image: self.expectedReplacementImage!, to: downscaledSize).cgImage,
+                let expectedCgImageData = expectedCgImage.dataProvider?.data,
+                let expectedCgImageBytes = CFDataGetBytePtr(expectedCgImageData) else {
+                XCTFail("Couldn't access CGImage data")
+                return
+            }
+            XCTAssert(gotCgImage.colorSpace?.model == .rgb)
```

Comment:
Just curious, why downscale and compare the size?

Reply:
Added a comment to make this clearer